### DISCUSSION TITLE
Is it possible to deploy skywalking from source code on kubernetes with helm scripts?

### DISCUSSION THREAD

**Discussion by Yangfisher1** [apache/skywalking#12167] [Q&A] [answered]
Hi, I'd like to know if it's possible to deploy SkyWalking on Kubernetes directly from its source code using Helm scripts, just like the [repository](https://github.com/apache/skywalking-helm). However, it only tells the user how to install `skywalking-helm` development version using the Helm script source codes, rather than the source codes of SkyWalking itself.

As far as I understand, I should modify the skywalking source code to generate a container image for it, and then change the image address in the helm chart so that it can be deployed?

**Comment by wu-sheng** [comment]
What are you saying? I am totally not following. Source codes are source codes. If you want to build image from source codes and use it, go ahead. There is nothing blocking you.

**Comment by Yangfisher1** [comment]
@wu-sheng Apologies for the bad description. 

What I mean is that this [repository](https://github.com/apache/skywalking-helm) only provides a tutorial for installing the released version of SkyWalking on Kubernetes, but does not offer a method for installing SkyWalking on Kubernetes from the source code.

**Reply by wu-sheng** [reply, marked as answer]
No, there isn't. We would not provide things like that. Private fork is OK according to APLv2, but how to fork, change and redistribute, that is your responsibility.

**Reply by Yangfisher1** [reply]
Got it, thanks for your help!
